tests/files/fips-test-gcrypt.c

```diff
@@ -0,0 +1,65 @@
+/** Simple C program that will calculate the hash of a fixed string using
+ * libgcrypt.
+ *
+ * The hash to be used is passed as the first parameter to the binary.
+ * This program demonstrates the use of libgcrypt for computing message digests.
+ */
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <gcrypt.h>
+
+int main(int argc, char *argv[]) {
+    gcry_md_hd_t md_hd;
+    const char *mess1 = "Test Message\n";
+    const char *mess2 = "Hello World\n";
+    unsigned char md_value[128];
+    int ret_code = 1;
+
+    if (argc < 2) {
+        fprintf(stderr, "Usage: fips-test-gcrypt digestname\n");
+        return ret_code;
+    }
+
+    if (!gcry_check_version(GCRYPT_VERSION)) {
+        fprintf(stderr, "libgcrypt version mismatch\n");
+        return ret_code;
+    }
+    gcry_control(GCRYCTL_INITIALIZATION_FINISHED, 0);
+
+    const int algo = gcry_md_map_name(argv[1]);
+    if (algo == 0) {
+        fprintf(stderr, "Unknown message digest %s\n", argv[1]);
+        goto cleanup;
+    }
```

Comment:
You can also return here immediately, as you haven't touched `md_hd` yet
```suggestion
    if (algo == 0) {
        fprintf(stderr, "Unknown message digest %s\n", argv[1]);
        return ret_code;
    }
```

Reply:
fixed